src/tools/debugger/FrontendConnector.java

```diff
@@ -266,6 +371,36 @@ public void shutdown() {
     try {
       int delayMsec = 1000;
       receiver.stop(delayMsec);
-    } catch (InterruptedException e) { }
+    } catch (InterruptedException e) {
+
+    }
+  }
+
+  /**
+   * Encapsulates the holder class and the method name of the breakpoint.
+   */
+  public class BreakpointLocation {
+    private String holderClass;
+    private String methodName;
+    private BreakpointId id;
+
+    BreakpointLocation(final String holderClass, final String methodName,
+        final BreakpointId id) {
+      this.holderClass = holderClass;
+      this.methodName = methodName;
```

Comment:
Thinking about this, I don't think we want to identify the method by class and name.
Instead, we should just identify it by its source location. A `Method` has a source section. That one identifies it uniquely. And, also avoids having two classes with the same name and the same method name clash.